### DISCUSSION TITLE
Backwards compatibility inquiry

### DISCUSSION THREAD

**Discussion by MattAtA3D** [desjarlais/Scintilla.NET#66] [Q&A] [answered]
Are the newer versions of this library drop-in compatible with the last release in [jacobslusser/ScintillaNET](https://github.com/jacobslusser/ScintillaNET)?

I'm using a commercial software package that is built against the 3.6.3.0 release, and I'm running into an `OutOfMemoryException` during certain operations because of a memory leak in `SciLexer`. If I can (with reasonable confidence) just drop in an updated version of `Scintilla[.]Net.dll` and go, that would definitely be the fastest path forward.

**Comment by VPKSoft** [comment, marked as answer]
Hi,
This should be backwards compatible. There is an unresolved issue of .NET Framework #52 with WinForms designer. Also the Lexer property will probably be dropped in the future as the 5.x series of the Scintilla will be moving to use names instead of constants to set the lexer to use.
I'm not giving any guarantees though.

**Reply by MattAtA3D** [reply]
> Hi, This should be backwards compatible. There is an unresolved issue of .NET Framework #52 with WinForms designer. Also the Lexer property will probably be dropped in the future as the 5.x series of the Scintilla will be moving to use names instead of constants to set the lexer to use.

Thanks for confirming that 👍 

> I'm not giving any guarantees though.

I wouldn't expect any. Just wanted to make sure it wasn't known to be a non-starter.

**Comment by desjarlais** [comment]
Closing this one out since it seems to be answered.  Although @MattAtA3D , if this needs more discussion, reopen and we can talk more about it.  

Nice profile image, Bluey is a favorite with my kids...although to be honest, I think I like it more than they do.
